### PR TITLE
arch: posix: posix_cheats: Improve it

### PR DESCRIPTION
**Pull request by aescolar**
Add a few comments in posix_cheats.h explaining why the file exists and how it is used.
There is no need to rename the POSIX types, so let's not do it to simplify things
Also remove an unnecessary guard (POSIX_ARCH) to avoid mystifying this any more than necessary

Related to #13054

### CODE REVIEW

**Comment by aescolar** [comment]
@pfalcon : Is this improving things a bit?

File: arch/posix/include/posix_cheats.h [hunk 1]
```diff
@@ -12,8 +12,6 @@
 #if !defined(ZEPHYR_ARCH_POSIX_INCLUDE_POSIX_CHEATS_H_) && !defined(NO_POSIX_CHEATS)
 #define ZEPHYR_ARCH_POSIX_INCLUDE_POSIX_CHEATS_H_
 
-#ifdef CONFIG_ARCH_POSIX
```
Comment (pfalcon) [inline arch/posix/include/posix_cheats.h]: Well, I'm not sure about removal of this guard. I'd rather have it stay, or worst-case, we'll have behavior changes outside ARCH_POSIX, whereas the aim to get it right for ARCH_POSIX.

Comment (pfalcon) [reply arch/posix/include/posix_cheats.h]: > Well, I'm not sure about removal of this guard.

Ok, now I see that comments say that this header is included only for ARCH_POSIX anyway.

**Review by pfalcon — APPROVED** [review]
@aescolar, Thanks for quickly coming out with the patch. Yes, this definitely explains or at least gives hints in the right direction where to look for more info.

**Review by nashif — APPROVED** [review]

**Pull request merged**